client/src/+assets/user-profile.tsx

```diff
@@ -0,0 +1,222 @@
+import { newLogin, StoreItemTypes } from '../shared'
+import { NotificationPopup } from '../components'
+import styles from './user-profile.module.scss'
+
+const b = BEM('user-profile', styles)
+
+interface AdditionalInformation {
+    linkedinProfile: string
+}
+
+interface UserProfileParams {
+    nickname: string
+    name?: string
+    email?: string
+    additionalInformation?: unknown
+}
+
+export const UserProfile: NextPage = () => {
+    const {sdk, account, loginMarketplaceAPI, web3 } = useContext(User)
+    const [inputError, setInputError] = useState('') 
+    const [errorMessage, setErrorMessage] = useState('')
+    const [successMessage, setSuccessMessage] = useState('')
+    const [isUpdated, setIsUpated] = useState(false)
+    const [isAddressAdded, setIsAddressAdded] = useState(false)
+    const popupRef = useRef<UiPopupHandlers>()
+
+    const [userId, setUserId] = useState('')
+    const [userProfile, setUserProfile] = useState<UserProfileParams>({
+        nickname: '',
+        name: '',
+        email: '',
+        additionalInformation: {
+            linkedinProfile: '',
+        },
+    })
+    const [newAddress, setNewAddress] = useState('')
+
+    const [addresses, setAddresses] = useState<string[]>([])
+    
+    const closePopup = (event: any) => {
+        popupRef.current?.close()
+        event.preventDefault()
+    }
+
+    const onAddAddress = async () => {
+        try {
+            const accounts = await sdk.accounts.list()
+            const accountToAdd = accounts.find(a => a.getId().toLowerCase() === newAddress)
+            console.log(accountToAdd)
+            const credential = await sdk.utils.jwt.generateClientAssertion(accountToAdd)
+            const token = await sdk.marketplace.addNewAddress(credential)
+            localStorage.setItem(StoreItemTypes.MarketplaceApiToken, token)
+            setAddresses([...addresses, newAddress])
+            setNewAddress('')
+            setIsAddressAdded(true)
+            setSuccessMessage('Added is added successfully')
+        } catch (error: any) {
+            if(error.message.includes('"statusCode":401')) {
+                setErrorMessage('Your login is expired. Please change to the previous address, reload and sign again')
+                localStorage.removeItem(StoreItemTypes.MarketplaceApiToken)
+            }
+
+            setErrorMessage(error.message)
+            popupRef.current?.open()
+        }
+        
+    }
+
+    const onSubmitUserProfile = async() => {
+        try {
+            if (!userProfile.nickname) {
+                setInputError('Nickname is required')
+                return
+            }
+            await sdk.profiles.update(userId, userProfile)
+            setIsUpated(true)
+            setSuccessMessage('Your profile is updated successfully')
+            setInputError('')
+        } catch (error: any ) {
+            if(error.message.includes('"statusCode":401')) {
+                newLogin(sdk, loginMarketplaceAPI)
```

Comment:
`onAddAddress` handles the 401 differently. it removes the token from localStorage, it doesn't call `newLogin` and has a different error message. Is this the expected behavior?

Reply:
yes because If I login with the new account wallet it will create another different User Profile and what we want is include it to the existing profile